### PR TITLE
fix: Fixed AuthService cookies expiration date bug. Added ability to disable redirect in AuthService unauthorize method

### PR DESCRIPTION
**Pull request by VladimirGlinskii**

### CODE REVIEW

File: common/src/auth/auth.service.ts [hunk 1]
```diff
@@ -88,7 +92,13 @@ export class AuthService<User extends AbstractUser> {
     this.resetRemember();
     this.userService.resetProfile();
 
-    this.router.navigate([this.authConfig.unauthenticatedRoute ?? AuthService.DEFAULT_UNAUTHENTICATED_ROUTE]);
+    const redirectRoute = (this.authConfig.unauthenticatedRoute === undefined)
+      ? AuthService.DEFAULT_UNAUTHENTICATED_ROUTE
+      : this.authConfig.unauthenticatedRoute
```
Comment (astorozhevsky) [inline common/src/auth/auth.service.ts:97]:
```suggestion
      : this.authConfig.unauthenticatedRoute;
```

File: common/src/auth/auth.service.ts [hunk 2]
```diff
@@ -88,7 +92,13 @@ export class AuthService<User extends AbstractUser> {
     this.resetRemember();
     this.userService.resetProfile();
 
-    this.router.navigate([this.authConfig.unauthenticatedRoute ?? AuthService.DEFAULT_UNAUTHENTICATED_ROUTE]);
+    const redirectRoute = (this.authConfig.unauthenticatedRoute === undefined)
```
Comment (astorozhevsky) [inline common/src/auth/auth.service.ts:95]: sorry, I'm not sure that it differs from the current implementation:

```ts
this.router.navigate([this.authConfig.unauthenticatedRoute ?? AuthService.DEFAULT_UNAUTHENTICATED_ROUTE]);
```

They don't seem to be any different, but the current implementation is much shorter.

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAJgUwGYEMCuAbKA1FG0IwC8MA5IqplKQNwBQoksAbngcTGmBQJZgJx6DcBBAYEAOgwgA5gApW+QgH5l8ZOiy4lAShpA

if we want to be able to disable the redirect, then I would suggest another implementation:

1. add a property `disableRedirectAfterUnauthorize?: false` in the config
2. check it here

File: common/src/cookie/cookie.service.ts [hunk 3]
```diff
@@ -55,6 +55,10 @@ export class CookieService<TKey extends string = string> {
     return this.parseCookieString(this.cookieString);
   }
 
+  public hasKey(key: TKey): boolean {
```
Comment (astorozhevsky) [inline common/src/cookie/cookie.service.ts:58]: it seems unused

**Review by astorozhevsky — CHANGES_REQUESTED** [review]

File: common/src/cookie/cookie.service.ts (hunk 3 above)
Comment (VladimirGlinskii) [reply common/src/cookie/cookie.service.ts:58]: @astorozhevsky It will be used in real apps.

**Review by astorozhevsky — APPROVED** [review]

**Pull request merged**